pyop2/codegen/rep2loopy.py

```diff
@@ -655,12 +649,21 @@ def statement_functioncall(expr, context):
 
     call = pym.Call(pym.Variable(expr.name), tuple(reads))
 
-    return loopy.CallInstruction(tuple(writes), call,
+    insn = loopy.CallInstruction(tuple(writes), call,
                                  within_inames=within_inames,
                                  predicates=predicates,
                                  id=id,
                                  depends_on=depends_on, depends_on_is_final=True)
 
+    # We cannot reconstruct the correct calling convention for C-string kernels
+    # without providing some additional context about the argument ordering.
+    # This is processed inside the ``emit_call_insn`` method of
+    # :class:`.PyOP2KernelCallable`.
+    if not hasattr(context, "function_args"):
+        context.function_args = {}
```

Comment:
Just set this up in the `WrapperBuilder` constructor?

Reply:
Now set up where `context` is defined.